table-api/src/main/java/io/deephaven/api/filter/FilterComparison.java

```diff
@@ -5,60 +5,71 @@
 
 import io.deephaven.annotations.BuildableStyle;
 import io.deephaven.api.ColumnName;
-import io.deephaven.api.value.Value;
+import io.deephaven.api.expression.Expression;
 import org.immutables.value.Value.Immutable;
 
 import java.io.Serializable;
+import java.util.Objects;
 
 /**
  * Evaluates to true based on the specific {@link #operator() operator} applied to the {@link #lhs() left-hand side} and
- * {@link #rhs() right-hand side}.
+ * {@link #rhs() right-hand side} {@link Expression expressions}.
  */
 @Immutable
 @BuildableStyle
-public abstract class FilterCondition extends FilterBase implements Serializable {
+public abstract class FilterComparison extends FilterBase implements Serializable {
 
     public enum Operator {
         /**
          * {@code lhs < rhs}
          */
-        LESS_THAN,
+        LESS_THAN("<"),
 
         /**
          * {@code lhs <= rhs}
          */
-        LESS_THAN_OR_EQUAL,
+        LESS_THAN_OR_EQUAL("<="),
 
         /**
          * {@code lhs > rhs}
          */
-        GREATER_THAN,
+        GREATER_THAN(">"),
 
         /**
          * {@code lhs >= rhs}
          */
-        GREATER_THAN_OR_EQUAL,
+        GREATER_THAN_OR_EQUAL(">="),
 
         /**
          * {@code lhs == rhs}
          */
-        EQUALS,
+        EQUALS("=="),
 
         /**
          * {@code lhs != rhs}
          */
-        NOT_EQUALS;
+        NOT_EQUALS("!=");
 
-        public final FilterCondition of(Value lhs, Value rhs) {
-            return FilterCondition.builder().operator(this).lhs(lhs).rhs(rhs).build();
+        private final String javaOperator;
+
+        Operator(String javaOperator) {
+            this.javaOperator = Objects.requireNonNull(javaOperator);
+        }
+
+        public String javaOperator() {
+            return javaOperator;
+        }
+
+        public FilterComparison of(Expression lhs, Expression rhs) {
+            return FilterComparison.builder().operator(this).lhs(lhs).rhs(rhs).build();
         }
 
         /**
          * The inverted, or negated, operator.
          *
          * @return the inverted operator
          */
-        public final Operator invert() {
+        public Operator inverse() {
```

Comment:
isn't a verb better than a noun/adjective for a method name?

Reply:
Oops, will do.